src/actions/guides/database/querying.cr

```diff
@@ -7,294 +7,526 @@ class Guides::Database::Querying < GuideAction
-      will always return no records. For example: `UserQuery.new.none.first` will never return a result
+    To do paginating, you'll use a combination of limit and offset. You can also use this formula to help.
 
-    ## Shorthand methods
+    ```crystal
+    page = 1
+    per_page = 12
 
-    There are a few methods that are used so often that we've added shortcut methods:
+    limit = per_page
+    offset = per_page * (page - 1)
 
-    * `find`
-    * `first`
-    * `last`
+    UserQuery.new.limit(limit).offset(offset)
+    ```
+
+    ### Limit
+
+    `SELECT COLUMNS FROM users LIMIT 1`
 
     ```crystal
-    id = 1
-    UserQuery.find(id)
-    UserQuery.first
-    UserQuery.last
+    UserQuery.new.limit(1)
     ```
 
-    ## Type specific query methods
+    ### Offset
+
+    `SELECT COLUMNS FROM users OFFSET 20`
+
+    ```crystal
+    UserQuery.new.offset(20)
+    ```
 
-    The previous section mostly focused on simple equality queries, but you can also
-    do more advanced queries depending on the column type.
+    ## Aggregates
 
-    ### All types
+    ### Count
 
-    * `is`
-    * `desc_order`
-    * `asc_order`
+    `SELECT COUNT(*) FROM users`
 
-    ### String
+    ```crystal
+    total_count = UserQuery.new.select_count
+    ```
 
-    * `like`
-    * `ilike`
-    * `lower`
+    ### Avg / Sum
 
-    ### Int32 and Time
+    `SELECT AVG(users.age) FROM users`
 
-    * `gt`
-    * `gte`
-    * `lt`
-    * `lte`
+    ```crystal
+    UserQuery.new.age.select_average
+    ```
 
-    ### Examples
+    `SELECT SUM(users.age) FROM users`
 
     ```crystal
-    UserQuery.new.age.gte(18).name.lower.is("sally")
+    UserQuery.new.age.select_sum
     ```
 
-    ## Using scopes
+    ### Min / Max
+
+    `SELECT MIN(users.age) FROM users`
 
-    Generally it's best to name your query methods so they are easy to understand
-    and reuse across your app. In Lucky, this is done by extracting methods on the
-    query object.
+    ```crystal
+    UserQuery.new.age.select_min
+    ```
 
-    Here's an example:
+    `SELECT MAX(users.age) FROM users`
 
     ```crystal
-    class UserQuery < User::BaseQuery
-      def adults
-        age.gte(18)
-      end
+    UserQuery.new.age.select_max
+    ```
 
-      def search(name)
-        ilike("\#{name}%")
-      end
-    end
+    ## Associations and Joins
 
-    UserQuery.new.adults.search("Sal")
+    When you have a model that is associated to another, your association is a method you can use
+    to return those records.
+
+    ### Associations
+
+    Each association defined on your model will have a method that takes a block, and passed in the
+    query for that association.
+
+    You can use this to help refine your association.
+
+    ```crystal
+    UserQuery.new.join_taks.tasks { |task_query|
+      # WHERE tasks.title = 'Clean up notes'
+      task_query.title("Clean up notes")
+    }
+    ```
```

Comment:
I think adding some text on what this would do in written word would help clarify too. Something like: 

> This will return all users who has a task with a title "Clean up notes"

Or  something like that